### PR TITLE
fix #11221: added version pages for missing entities

### LINKED ISSUE #11221: UI - add version pages for missing entities
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**

- [x] The version page for Topic details is not showing all the details refer screenshot:

image

- [x] Topic config should show all the config available from backend instead filter it out based on the Top bar available

**To Reproduce**

1. Refer version page for any topic and check history

**Expected behavior**

- It should show version details for all others tabs as well

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes:

Fixes #11221 

I worked on adding version pages for the following entities

- [x] Database

https://github.com/open-metadata/OpenMetadata/assets/51777795/337e72d2-1878-4cd1-bad6-2a42210797b1

- [x] Database Schema

https://github.com/open-metadata/OpenMetadata/assets/51777795/09646436-5ed1-406b-8768-ad31e4f41324

- [x] Classifications

https://github.com/open-metadata/OpenMetadata/assets/51777795/9aa3fb54-7db8-49b8-8df2-33efea78cb95

#
### Type of change:

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx [hunk 1]
```diff
@@ -36,21 +34,22 @@ interface SchemaTablesTabProps {
...
   tableDataLoading: boolean;
   description: string;
-  entityFieldThreadCount: EntityLinkThreadCount[];
-  editDescriptionPermission: boolean;
-  isEdit: boolean;
-  showDeletedTables: boolean;
+  entityFieldThreadCount?: EntityLinkThreadCount[];
+  editDescriptionPermission?: boolean;
+  isEdit?: boolean;
+  showDeletedTables?: boolean;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:40]: @aniketkatkar97 , why we are making these props optional? if we are doing so then we should have default values or at least it should be guarded.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+      return <Loader />;
+    }
+
+    if (!viewVersionPermission) {
+      return <ErrorPlaceHolder type={ERROR_PLACEHOLDER_TYPE.PERMISSION} />;
+    }
+
+    return (
+      <>
+        {isVersionDataLoading ? (
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx:297]: Shouldn't this check at the top with isLoading?

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    }
+
+    if (!viewVersionPermission) {
+      return <ErrorPlaceHolder type={ERROR_PLACEHOLDER_TYPE.PERMISSION} />;
+    }
+
+    return (
+      <>
+        {isVersionDataLoading ? (
+          <Loader />
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx:314]: Same here.

File: openmetadata-ui/src/main/resources/ui/src/utils/Database/DatabaseDetails.utils.tsx
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/Database/DatabaseDetails.utils.tsx:1]: Let's move this file to the utils level and remove the database folder.

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityVersionUtils.tsx [hunk 4]
```diff
@@ -218,7 +220,13 @@ export const getTagsDiff = (
...
 
 export const getEntityVersionTags = (
-  currentVersionData: VersionData | Glossary | GlossaryTerm | ServicesType,
+  currentVersionData:
+    | VersionData
+    | Glossary
+    | GlossaryTerm
+    | ServicesType
+    | Database
+    | DatabaseSchema,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityVersionUtils.tsx:229]: Create a type for it and use it here.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    }
+  }, [currentVersionData]);
+
+  return (
+    <PageLayoutV1
+      className="version-page-container"
+      pageTitle={t('label.entity-version-detail-plural', {
+        entity: getEntityName(currentVersionData),
+      })}>
+      {versionComponent()}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx:367]: Can we avoid rendering through function, If function doesn't based on any param

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+  const tier = useMemo(
+    () => getTierTags(currentVersionData.tags ?? []),
+    [currentVersionData]
+  );
+  const owner = useMemo(() => currentVersionData.owner, [currentVersionData]);
+
+  const breadcrumbLinks = useMemo(
+    () => getEntityBreadcrumbs(currentVersionData, EntityType.DATABASE),
+    [currentVersionData]
+  );
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx:99]: Can we use single useMemo?

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    return getEntityVersionByField(
+      changeDescription,
+      EntityField.DISPLAYNAME,
+      currentVersionData.displayName
+    );
+  }, [currentVersionData, changeDescription]);
+
+  const tags = useMemo(() => {
+    return getEntityVersionTags(currentVersionData, changeDescription);
+  }, [currentVersionData, changeDescription]);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx:226]: same

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+  const versionHandler = useCallback(
+    (newVersion = version) => {
+      history.push(getDatabaseVersionPath(databaseFQN, toString(newVersion)));
+    },
+    [databaseFQN]
+  );
+
+  const backHandler = useCallback(() => {
+    history.push(getDatabaseDetailsPath(databaseFQN));
+  }, [databaseFQN]);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx:214]: same

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+    }
+  }, [currentVersionData]);
+
+  return (
+    <PageLayoutV1
+      className="version-page-container"
+      pageTitle={t('label.entity-version-detail-plural', {
+        entity: getEntityName(currentVersionData),
+      })}>
+      {versionComponent()}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseVersionPage/DatabaseVersionPage.tsx:383]: same

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx [hunk 10]
```diff
@@ -192,14 +184,13 @@ function ServiceMainTabContent({
...
                   columns={tableColumn}
-                  components={tableComponent}
                   data-testid="service-children-table"
                   dataSource={data}
                   locale={{
                     emptyText: <ErrorPlaceHolder className="m-y-md" />,
                   }}
                   pagination={false}
-                  rowKey="id"
+                  rowKey="name"
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:193]: What was the issue with `id` based rowKey?

File: openmetadata-ui/src/main/resources/ui/src/components/ClassificationDetails/ClassificationDetails.tsx [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+      !isUndefined(currentClassification) &&
+      !isUndefined(handleUpdateClassification)
+    ) {
+      return handleUpdateClassification({
+        ...currentClassification,
+        ...data,
+      });
+    }
+
+    return new Promise<void>((resolve) => resolve());
```
[the remark this answers is not in the document]
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/ClassificationDetails/ClassificationDetails.tsx:178]: Did we address this?

File: openmetadata-ui/src/main/resources/ui/src/components/ClassificationDetails/ClassificationDetails.tsx [hunk 12]
```diff
@@ -0,0 +1,526 @@
...
+
+  const description = useMemo(() => {
+    return isVersionView
+      ? getEntityVersionByField(
+          changeDescription,
+          EntityField.DESCRIPTION,
+          currentClassification?.description
+        )
+      : currentClassification?.description;
+  }, [currentClassification, changeDescription]);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/ClassificationDetails/ClassificationDetails.tsx:357]: same

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaVersionPage/DatabaseSchemaVersionPage.tsx:297]: No, actually I kept it inside as there is a right panel that is independent of that.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**